tree/tree/src/TTree.cxx

```diff
@@ -8778,6 +8782,7 @@ Int_t TTree::SetCacheSizeAux(bool autocache /* = true */, Long64_t cacheSize /*
          if (res < 0) {
             return -1;
          }
+         cacheSize = pf->GetBufferSize(); // update after potential clamp
```

Comment:
@pcanal this is the line I was referring to. a cacheSize is requested which then has nothing to do with the actual cacheSize set because of the inherent clamping, thus an update of the value is needed for consistency, otherwise one gets unrealistic results. That's why the roottest had to be adapted.